### PR TITLE
Document comparing Mbed OS 2 and Mbed OS 5

### PR DESCRIPTION
**Pull request by yennster**
This is a document detailing the differences and similarities between Mbed OS 2 and Mbed OS 5. It also includes a FAQ section with the most frequently asked questions about Mbed OS 2 vs. Mbed OS 5 from community forum/question sections on the os.mbed.com website.

### CODE REVIEW

File: docs/tutorials/mbed_os_2_vs_5.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+## Arm Mbed OS 2 vs. Arm Mbed OS 5
+
+Arm Mbed OS 5 is a superset of Arm Mbed OS 2. This means that Mbed OS 5 *is* Mbed OS 2 but with more features, such as an RTOS.
+
+### Overview
+
+- Mbed OS 5 is a *superset* of Mbed OS 2.
+- Mbed OS 5 is a combination of both Mbed OS 2 and Mbed-RTOS.
+- Mbed OS 2 and Mbed OS 5 both use the same `.h` file.
```
Comment (theotherjimmy) [inline docs/tutorials/mbed_os_2_vs_5.md]: Another big difference worth mentioning: Drivers are thread-safe in Mbed OS 5, and are not thread-safe in Mbed OS 2. This implies that: Mbed OS 5 is more than Mbed OS 2 and Mbed-RTOS

File: docs/tutorials/mbed_os_2_vs_5.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+### FAQs
+
+**Do Mbed OS 2 and Mbed OS 5 use the same header file?**
```
Comment (theotherjimmy) [inline docs/tutorials/mbed_os_2_vs_5.md]: I think there should be a `Yes` somewhere in this section.

File: docs/tutorials/mbed_os_2_vs_5.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+   - [Mbed OS 2 Handbook](https://os.mbed.com/handbook/Homepage).
+   - [Mbed OS 2 Library](https://os.mbed.com/users/mbed_official/code/mbed/).
+
+   If you come across a documentation page for Mbed OS 2, and you would like to see if there is an updated version of this page for Mbed OS 5, from the Handbook page click on the link provided in the top yellow box. For example, you can see the yellow box at the top of the [LocalFileSystem page](https://os.mbed.com/handbook/LocalFileSystem) for Mbed OS 2.
+
+   <span class="images">![](https://os.mbed.com/media/uploads/jplunkett/mbedhandbookbox.png)</span>
+
+   Click on "the file system API" link to navigate to the updated documentation regarding this API for Mbed OS 5.
+
+**How do I know if my project uses Mbed OS 2 or Mbed OS 5?**
```
Comment (theotherjimmy) [inline docs/tutorials/mbed_os_2_vs_5.md]: You should point out something that "signifies that the project uses Mbed OS 5" (presence of `mbed-os.lib` is a good indicator)

**Comment by yennster** [comment]
@theotherjimmy Done! 😇

**Comment by AnotherButler** [comment]
@yennster Could you please upload these images to the "images" folder at the top of the docs structure? Thanks so much.

**Comment by AnotherButler** [comment]
Oops. Accidentally merged this to 5.7 instead of the test site first. Recreating against new_engine. Please see https://github.com/ARMmbed/Handbook/pull/431

**Comment by yennster** [comment]
@AnotherButler I just realized I messed up an image link (https://github.com/ARMmbed/Handbook/pull/393/commits/fa1beaadbc1085126225f06d1330a0e8a2fb3443#diff-13499430887824fbea1567b9d706c5d9R66) should I make a new PR or add it to the Migration Guide PR (https://github.com/ARMmbed/Handbook/pull/392)?

**Comment by yennster** [comment]
@AnotherButler Actually I'll just make a new PR so I don't have to mess with rebasing

**Comment by AnotherButler** [comment]
@yennster Sounds great, thanks 👍

**Comment by yennster** [comment]
@AnotherButler Image fixed in https://github.com/ARMmbed/Handbook/pull/444

**Pull request merged**